lib/cocoapods/target/pod_target.rb

```diff
@@ -600,7 +600,11 @@ def header_search_paths(include_test_dependent_targets = false)
       dependent_targets = recursive_dependent_targets
       dependent_targets += recursive_test_dependent_targets if include_test_dependent_targets
       dependent_targets.each do |dependent_target|
-        header_search_paths.concat(sandbox.public_headers.search_paths(platform, dependent_target.pod_name, defines_module? && dependent_target.uses_modular_headers?(false)))
+        header_search_paths.concat(sandbox.public_headers.search_paths(platform,
+                                                                       dependent_target.pod_name,
+                                                                       defines_module? &&
+                                                                         dependent_target.uses_modular_headers?(false) &&
+                                                                         dependent_target.spec_consumers.none?(&:header_dir)))
```

Comment:
please roll this into `uses_modular_headers?` right next to the check for `header_mappings_dir`, thanks!